test/java/controllers/KairosDbProxyControllerTest.java

```diff
@@ -194,4 +198,50 @@ public void testQueryRequiresAggregatorOff() {
         assertEquals(Http.Status.OK, result.status());
         assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
     }
+
+    @Test
+    public void testAddMergeAggregator() {
+        final Metric.Builder metric1Builder = new Metric.Builder()
+                .setName("metric1")
+                .setAggregators(ImmutableList.of(
+                        new Aggregator.Builder().setName("filter").build(),
+                        new Aggregator.Builder().setName("max").setSampling(
+                                new Sampling.Builder().setValue(1).setUnit(SamplingUnit.HOURS).build()).build()));
+        final Metric.Builder metric2Builder = new Metric.Builder()
+                .setName("metric2")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("min").build()));
+        final MetricsQuery.Builder builder = new MetricsQuery.Builder()
+                .setStartTime(Instant.now())
+                .setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
```

Comment:
It looks to me like the only way these builders are used is by calling `.build()` on them -- instead of
`final MetricsQuery.Builder builder = (...)`, why not write
`final MetricsQuery query = (...).build()`?

(I'm guessing that you copied this pattern from another test which declared some Builders; that's a reasonable pattern if you want to create several _variations_ on an object, since that lets you say
```java
foo = builder.setName("foo").build();
bar = builder.setName("bar").build();
baz = builder.setName("baz").build();
...
```
but if you're just building a single object, it's clearer to simply build the object up-front.)

Reply:
I see your point! But the previous test `testQueryRequiresAggregatorOff` also uses this pattern even though it also builds a single object. If I need to change this, should I change `testQueryRequiresAggregatorOff` as well?